cmd/nvidia-ctk/hook/chmod/chmod.go

```diff
@@ -126,17 +125,16 @@ func (m command) run(c *cli.Context, cfg *config) error {
 		return nil
 	}
 
-	locator := lookup.NewExecutableLocator(m.logger, "")
-	targets, err := locator.Locate("chmod")
-	if err != nil {
-		return fmt.Errorf("failed to locate chmod: %v", err)
+	for _, path := range paths {
+		err = os.Chmod(path, desiredMode)
```

Comment:
Just for my own information: What is the difference between this and `Exec`ing to `chmod`?

Reply:
I've tried to explain it in the commit message:

> Due to loosing control of the program after `exec()`-ing `chmod(1)` program
> and therefore not being able to handle errors -
> refactor to use `chmod(2)` syscall instead of `exec()` `chmod(1)` program.

Essentially, `nvidia-ctk` code stops executing and control is transferred fully to `chmod(1)` which will `exit(1)` once an error occurs.

Reply:
Yes, I'm aware of that. I meant if there was any difference in behaviour between the different `chmods`? Assuming that they were each called with a single path?

Reply:
chmod(1) does some checks and so on, but for our use-case it looks to be effectively the same.. or at least I hope so.
One thing I can think of: since 'mode' string was used as is - it could be in any form `chmod(1)` accepts (i.e. symbolic); but since we did not use that - should not be a problem.